internal/elasticsearch/common_test.go

```diff
@@ -564,6 +565,28 @@ func TestPodSpecHasTaintTolerations(t *testing.T) {
 	}
 }
 
+func TestElasticSearchSecurityContext(t *testing.T) {
+	podTemplate := newPodTemplateSpec(context.Background(), log.NewLogger("common-testing"), "test-node-name", "test-cluster-name", "test-namespace-name", api.ElasticsearchNode{}, api.ElasticsearchNodeSpec{}, map[string]string{}, map[api.ElasticsearchNodeRole]bool{}, nil, LogConfig{})
+
+	if !*podTemplate.Spec.SecurityContext.RunAsNonRoot {
+		t.Errorf("SecurityContext.RunAsNonRoot: expected true, actual %v", *podTemplate.Spec.SecurityContext.RunAsNonRoot)
+	}
+
+	expected := v1.SecurityContext{
+		ReadOnlyRootFilesystem:   pointer.BoolPtr(true),
+		AllowPrivilegeEscalation: pointer.Bool(false),
+		Capabilities: &v1.Capabilities{
+			Drop: []v1.Capability{"ALL"},
+		},
+	}
+
+	for _, c := range podTemplate.Spec.Containers {
+		if reflect.DeepEqual(c.SecurityContext, expected) {
+			t.Errorf("Container.SecurityContext: expected %+v, actual %v", expected, c.SecurityContext)
+		}
```

Comment:
Using `cmp.Diff` for comparing structs generates a more readable output on test failure. Bonus point you don't need to use the formatting stuff for the output, it's all inclusive.
```go
if diff := cmp.Diff(c.SecurityContext, expecte); diff != "" {
    t.Errorf("diff: %s", diff)
}
````